main/medium/src/audio_hook_register.rs

```diff
@@ -58,7 +59,35 @@ impl AudioHookRegister {
 
     /// Returns the current number of output channels.
     pub fn output_nch(&self) -> u32 {
-        unsafe { self.0.as_ref() }.input_nch as u32
+        unsafe { self.0.as_ref() }.output_nch as u32
+    }
+
+    /// Get access to the underlying samples of an output channel
+    pub fn output_channel_samples(&self, ch: usize, args: &OnAudioBufferArgs) -> Option<&mut [ReaSample]> {
```

Comment:
Instead of passing the `OnAudioBufferArgs` parameter, I think it would make sense to make `get_buffer()` a method of the `OnAudioBufferArgs` struct itself - because it contains a reference to the `AudioHookRegister` and therefore has all information necessary to implement the method.

The final signature in `OnAudioBufferArgs` would look like this:

`pub fn get_buffer(&self, kind: BufferKind, ch: u32) -> Option<&mut [ReaSample]>`

Reply:
I'll try it this way and update the PR.